pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptionsUtils.java

```diff
@@ -59,6 +59,17 @@ public static boolean isSkipUpsert(Map<String, String> queryOptions) {
     return Boolean.parseBoolean(queryOptions.get(Request.QueryOptionKey.SKIP_UPSERT));
   }
 
+  public static Integer getNumReplicaGroups(Map<String, String> queryOptions) {
+    String numReplicaGroups = queryOptions.get(Request.QueryOptionKey.NUM_REPLICA_GROUPS);
+    if (numReplicaGroups != null) {
+      int replicaGroups = Integer.parseInt(numReplicaGroups);
+      Preconditions.checkState(replicaGroups > 0, "numReplicaGroups must be positive, got: %s", numReplicaGroups);
+      return replicaGroups;
+    } else {
+      return null;
```

Comment:
Integer.parseInt() throws exception if the provided string is not a number. Can we treat null string the same way - not a number exception - and not special case it? That way the method is slightly simpler; the function return value can be primitive int; and there's no need on the calling function to check for null values.

Reply:
@sajjad-moradi I think we need the `null` return value to indicate that this option is not set, and create the routing table in the default way